diff --git a/cpp/src/arrow/filesystem/api.h b/cpp/src/arrow/filesystem/api.h
index 732be5f92..adadf6a51 100644
--- a/cpp/src/arrow/filesystem/api.h
+++ b/cpp/src/arrow/filesystem/api.h
@@ -20,6 +20,9 @@
 #include "arrow/util/config.h"  // IWYU pragma: export

 #include "arrow/filesystem/filesystem.h"  // IWYU pragma: export
+#ifdef ARROW_AZURE
+#include "arrow/filesystem/azurefs.h"  // IWYU pragma: export
+#endif
 #include "arrow/filesystem/hdfs.h"        // IWYU pragma: export
 #ifdef ARROW_GCS
 #include "arrow/filesystem/gcsfs.h"  // IWYU pragma: export
diff --git a/cpp/src/arrow/filesystem/type_fwd.h b/cpp/src/arrow/filesystem/type_fwd.h
index 892f7ad2e..c56baf79f 100644
--- a/cpp/src/arrow/filesystem/type_fwd.h
+++ b/cpp/src/arrow/filesystem/type_fwd.h
@@ -47,6 +47,7 @@ class SlowFileSystem;
 class LocalFileSystem;
 class S3FileSystem;
 class GcsFileSystem;
+class AzureBlobFileSystem;

 }  // namespace fs
 }  // namespace arrow
diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index bad5e926a..08840740d 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -424,6 +424,10 @@ if(PYARROW_BUILD_S3)
   set(CYTHON_EXTENSIONS ${CYTHON_EXTENSIONS} _s3fs)
 endif()

+if(PYARROW_BUILD_AZURE)
+  set(CYTHON_EXTENSIONS ${CYTHON_EXTENSIONS} _azurefs)
+endif()
+
 if(PYARROW_BUILD_HDFS)
   set(CYTHON_EXTENSIONS ${CYTHON_EXTENSIONS} _hdfs)
 endif()
diff --git a/python/pyarrow/__init__.py b/python/pyarrow/__init__.py
index bcb6b30e7..a02739e00 100644
--- a/python/pyarrow/__init__.py
+++ b/python/pyarrow/__init__.py
@@ -145,7 +145,7 @@ def show_info():
         print(f"  {module: <20}: {status: <8}")

     print("\nFilesystems:")
-    filesystems = ["GcsFileSystem", "HadoopFileSystem", "S3FileSystem"]
+    filesystems = ["GcsFileSystem", "HadoopFileSystem", "S3FileSystem", "AzureBlobFileSystem"]
     for fs in filesystems:
         status = "Enabled" if _filesystem_is_available(fs) else "-"
         print(f"  {fs: <20}: {status: <8}")
