diff --git a/reprotest/build.py b/reprotest/build.py
index 52eff03..fa5140c 100644
--- a/reprotest/build.py
+++ b/reprotest/build.py
@@ -302,7 +302,7 @@ def home(ctx, build, vary):
 def kernel(ctx, build, vary):
     _ = build
     if not vary:
-        _ = _.append_setup_exec_raw('SETARCH_ARCH=$(uname -m)')
+        return _
     else:
         _ = _.append_setup_exec_raw('SETARCH_ARCH=$(for a in $(setarch --list); do setarch $a true && echo $a || true; done)')
         # Perform realistic shuffling of architectures depending
