diff --git a/include/aws/common/byte_order.inl b/include/aws/common/byte_order.inl
index 1204be06a..d485ac38b 100644
--- a/include/aws/common/byte_order.inl
+++ b/include/aws/common/byte_order.inl
@@ -35,10 +35,8 @@ AWS_STATIC_IMPL uint64_t aws_hton64(uint64_t x) {
     if (aws_is_big_endian()) {
         return x;
     }
-#if defined(__x86_64__) && (defined(__GNUC__) || defined(__clang__)) && !defined(CBMC)
-    uint64_t v;
-    __asm__("bswap %q0" : "=r"(v) : "0"(x));
-    return v;
+#if (defined(__GNUC__) || defined(__clang__)) && !defined(CBMC)
+    return __builtin_bswap64(x);
 #elif defined(_MSC_VER)
     return _byteswap_uint64(x);
 #else
