diff --git a/ports/curl/portfile.cmake b/ports/curl/portfile.cmake
index 5a14562..924b1b7 100644
--- a/ports/curl/portfile.cmake
+++ b/ports/curl/portfile.cmake
@@ -87,8 +87,11 @@ vcpkg_cmake_configure(
         -DENABLE_MANUAL=OFF
         -DCURL_CA_FALLBACK=ON
         -DCURL_USE_LIBPSL=OFF
+        -DCURL_CA_PATH=none
+        -DCURL_CA_BUNDLE=none
     OPTIONS_DEBUG
         -DENABLE_DEBUG=ON
+        ${EXTRA_ARGS_DEBUG}
 )
 vcpkg_cmake_install()
 vcpkg_copy_pdbs()
diff --git a/ports/snappy/portfile.cmake b/ports/snappy/portfile.cmake
index df95a08..d740ce7 100644
--- a/ports/snappy/portfile.cmake
+++ b/ports/snappy/portfile.cmake
@@ -9,6 +9,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         fix_clang-cl_build.patch
+        "snappy-disable-bmi.patch"
 )
 
 vcpkg_cmake_configure(
diff --git a/ports/snappy/snappy-disable-bmi.patch b/ports/snappy/snappy-disable-bmi.patch
new file mode 100644
index 0000000000..a57ce0c22f
--- /dev/null
+++ b/ports/snappy/snappy-disable-bmi.patch
@@ -0,0 +1,19 @@
+diff --git a/snappy.cc b/snappy.cc
+index 79dc0e8..f3153ea 100644
+--- a/snappy.cc
++++ b/snappy.cc
+@@ -965,14 +965,10 @@ static inline void Report(const char *algorithm, size_t compressed_size,
+ static inline uint32_t ExtractLowBytes(uint32_t v, int n) {
+   assert(n >= 0);
+   assert(n <= 4);
+-#if SNAPPY_HAVE_BMI2
+-  return _bzhi_u32(v, 8 * n);
+-#else
+   // This needs to be wider than uint32_t otherwise `mask << 32` will be
+   // undefined.
+   uint64_t mask = 0xffffffff;
+   return v & ~(mask << (8 * n));
+-#endif
+ }
+ 
+ static inline bool LeftShiftOverflows(uint8_t value, uint32_t shift) {
diff --git a/scripts/cmake/vcpkg_find_acquire_program.cmake b/scripts/cmake/vcpkg_find_acquire_program.cmake
index 4611af6..d11936f 100644
--- a/scripts/cmake/vcpkg_find_acquire_program.cmake
+++ b/scripts/cmake/vcpkg_find_acquire_program.cmake
@@ -239,7 +239,7 @@ function(vcpkg_find_acquire_program program)
             set(paths_to_search "${DOWNLOADS}/tools/python/${tool_subdirectory}")
             vcpkg_list(SET post_install_command "${CMAKE_COMMAND}" -E rm python310._pth)
         else()
-            set(program_name python3)
+            set(program_name python)
             set(brew_package_name "python")
             set(apt_package_name "python3")
         endif()
