diff --git a/ports/curl/portfile.cmake b/ports/curl/portfile.cmake
index 6788bc7b7f..6b689dedf0 100644
--- a/ports/curl/portfile.cmake
+++ b/ports/curl/portfile.cmake
@@ -83,10 +83,13 @@ vcpkg_cmake_configure(
         -DENABLE_CURL_MANUAL=OFF
         -DIMPORT_LIB_SUFFIX=   # empty
         -DSHARE_LIB_OBJECT=OFF
+        -DCURL_CA_PATH=none
+        -DCURL_CA_BUNDLE=none
         -DCURL_USE_PKGCONFIG=ON
         -DCMAKE_DISABLE_FIND_PACKAGE_Perl=ON
     MAYBE_UNUSED_VARIABLES
         PKG_CONFIG_EXECUTABLE
+        ${EXTRA_ARGS_DEBUG}
 )
 vcpkg_cmake_install()
 vcpkg_copy_pdbs()
diff --git a/ports/llvm/portfile.cmake b/ports/llvm/portfile.cmake
index 7764357a6d..da7374ecec 100644
--- a/ports/llvm/portfile.cmake
+++ b/ports/llvm/portfile.cmake
@@ -302,6 +302,8 @@ vcpkg_cmake_configure(
         ${FEATURE_OPTIONS}
     MAYBE_UNUSED_VARIABLES 
         COMPILER_RT_ENABLE_IOS
+        BOLT_TOOLS_INSTALL_DIR
+        LIBOMP_INSTALL_ALIASES
 )
 
 vcpkg_cmake_install(ADD_BIN_TO_PATH)
diff --git a/ports/orc/orc-fix-exception-propagation.diff b/ports/orc/orc-fix-exception-propagation.diff
new file mode 100644
index 0000000000..25568e70cd
--- /dev/null
+++ b/ports/orc/orc-fix-exception-propagation.diff
@@ -0,0 +1,30 @@
+diff --git a/c++/src/Timezone.cc b/c++/src/Timezone.cc
+index 384f8ea99..07c75e0a7 100644
+--- a/c++/src/Timezone.cc
++++ b/c++/src/Timezone.cc
+@@ -696,12 +696,21 @@ namespace orc {
+     std::string filename_;
+     mutable std::unique_ptr<TimezoneImpl> impl_;
+     mutable std::once_flag initialized_;
++    mutable std::exception_ptr init_exception_;
+ 
+     TimezoneImpl* getImpl() const {
+-      std::call_once(initialized_, [&]() {
+-        auto buffer = loadTZDB(filename_);
+-        impl_ = std::make_unique<TimezoneImpl>(filename_, std::move(buffer));
+-      });
++        std::call_once(initialized_, [&]() {
++          try {
++            auto buffer = loadTZDB(filename_);
++            impl_ = std::make_unique<TimezoneImpl>(filename_, std::move(buffer));
++          } catch (...) {
++            // If initialization failed, re-throw the exception
++            init_exception_ = std::current_exception();
++          }
++        });
++      if (init_exception_) {
++        std::rethrow_exception(init_exception_);
++      }
+       return impl_.get();
+     }
+ 
diff --git a/ports/orc/portfile.cmake b/ports/orc/portfile.cmake
index 77ebf41ec3..4d065594a7 100644
--- a/ports/orc/portfile.cmake
+++ b/ports/orc/portfile.cmake
@@ -6,6 +6,8 @@ vcpkg_from_github(
   REF "v${VERSION}"
   SHA512 eabee16a6e984452a8cb715d0524041b20dd1bd88d78bb32534db93e5dbdd786aa4df8c05975406cb0728241eb3025a506c4fefb8c334ef0d8a27e6cb920d44c
   HEAD_REF master
+  PATCHES
+    orc-fix-exception-propagation.diff
 )
 
 file(REMOVE "${SOURCE_PATH}/cmake_modules/FindGTest.cmake")
